tools/serverless-plugin/src/generators/service/notification-files/src/utils/aws-formatting.ts.template

```diff
@@ -31,6 +31,37 @@ export function extractNameFromResourceName(resourceName: string) {
     return resourceName.split('-').pop() as string;
 }
 
+/**
+ * Extracts the error details from the error detail object
+ * @param {NotifyError.InvocationEvent['detail']} detail - The detail object to extract the error details from
+ * @returns {NotifyError.Cause} The error details
+ */
+export function extractErrorDetails(
+    event: NotifyError.InvocationEvent
+): NotifyError.Cause {
+    const { cause, error } = event.detail;
+
+    try {
+        const { errorType, errorMessage } = JSON.parse(cause);
+        return {
+            errorType,
+            errorMessage,
+        };
+    } catch (err) {
+        // StepFunctions "Fail" steps return the error details in the "cause" field
+        // where as Lambda will include a JSON object in the "cause" field that
+        // contains the error details
+        if (event.source === 'aws.states') {
```

Comment:
Consider modelling errors as named classes to make the inevitable if/else explosion more readable. Should be some examples in the newer client repos? @kai-nguyen-aligent what do you think? I'm not a huge fan of 'functional' catch steps like this.

Alternatively we should look in to more robust validation/parsing for this section in the future. I don't have the context around how we use this code though.

Reply:
The errors themselves are entirely service-controlled, this specifically just ensures we pass the error on verbatim. We should probably consider a handbook entry for modelling errors in a way that makes sense both from a debugging in AWS and reading via these notifications perspective.